app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/SplashScreenActivity.java

```diff
@@ -135,26 +127,25 @@ public void unzipFile() {
         SharedPreferences.Editor editor = sharedPref.edit();
         try {
             for (DownloadExpansionFile.XAPKFile xf : xAPKS) {
-                expansionFilePath = getExpansionFilePath(xf.mIsMain, xf.mFileVersion);
-                expansionFile = new File(expansionFilePath);
-                expansionZipFile = new ZipFile(expansionFile);
-                _zip = new Zip(expansionZipFile, this);
-                unzipFilePath = getUnzippedExpansionFilePath();
-                packageNameDir = new File(unzipFilePath);
-                if (xf.mIsMain) {
-                    if (packageNameDir.exists()) {
-                        DownloadExpansionFile.deleteDir(packageNameDir);
+                if (xf.mIsMain && xf.mFileVersion != storedMainFileVersion || !xf.mIsMain && xf.mFileVersion != storedPatchFileVersion) {
+                    expansionFilePath = getExpansionFilePath(xf.mIsMain, xf.mFileVersion);
+                    expansionFile = new File(expansionFilePath);
+                    expansionZipFile = new ZipFile(expansionFile);
+                    _zip = new Zip(expansionZipFile, this);
+                    unzipFilePath = getUnzippedExpansionFilePath();
+                    packageNameDir = new File(unzipFilePath);
+                    if (xf.mIsMain && !packageNameDir.exists()) {
+                        packageNameDir.mkdir();
+                    }
+                    _zip.unzip(unzipFilePath, totalZipSize);
+                    _zip.close();
+                    if (xf.mIsMain) {
+                        editor.putInt(getString(R.string.mainFileVersion), xf.mFileVersion);
+                        editor.commit();
```

Comment:
Put editor.commit() outside if else and use editor.apply() as suggested by code analyzer

Reply:
Removed redundant `editor.commit()` statement. However, I will still stick to `editor.commit()` as mentioned above. Additionally, FYI, the `editor.apply()` statement saves the value in-memory instead of storing it persistently on a storage. It performs an asynchronous update to the persistent storage and does not notify about any failure.